include/boost/url/impl/url_base.ipp

```diff
@@ -270,20 +270,20 @@ set_user(string_view s)
 
 url_base&
 url_base::
-set_user(pct_encoded_view s)
+set_user(decode_view s)
 {
     grammar::detail::copied_strings<
         BOOST_URL_STACK_BYTES> buf(
             this->string());
     s = s.maybe_copy(buf);
     check_invariants();
     auto const n =
-        detail::pct_encode_bytes_impl(
+        detail::encoded_size_impl(
```

Comment:
Why aren't we using the public function?

Reply:
Same as the other one. We don't have an overload for arbitrary iterators and we can't use `s.encoded().size()` because it's a different `allowed` charset.

Reply:
So `s` is `decode_view` here?

Reply:
Yes. This is the `decode_view` overload.